### PR TITLE
feat: add `ai_to_sql` transalte natual lanauge to SQL based on your table schema

### PR DESCRIPTION
**Pull request by BohuTANG**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

```
CREATE DATABASE openai;
USE openai;

CREATE TABLE users(
    id INT,
    name VARCHAR,
    age INT
);

SELECT * FROM ai_to_sql('List all users older than 30 years', '<openai-api-key>');
```

Output:
```
*************************** 1. row ***************************
     database: simple_example
generated_sql:  SELECT * FROM users WHERE age > 30;
```

Closes #issue

### CODE REVIEW

File: src/query/service/Cargo.toml [hunk 1]
```diff
@@ -128,6 +129,7 @@ pin-project-lite = "0.2.9"
 poem = { version = "1", features = ["rustls", "multipart", "compression"] }
 rand = "0.8.5"
 regex = "1.6.0"
+reqwest = { version = "0.11.14", features = ['__rustls'] }
```
Comment (Xuanwo) [inline src/query/service/Cargo.toml:132]: Not needed, we can use the version specified in workspace.

File: docs/doc/15-sql-functions/61-ai-functions/index.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+CREATE DATABASE openai;
+USE openai;
+
+CREATE TABLE users(
+    id INT,
+    name VARCHAR,
+    age INT
+);
+
+SELECT * FROM ai_to_sql('List all users older than 30 years', '<openai-api-key>');
```
Comment (sundy-li) [inline docs/doc/15-sql-functions/61-ai-functions/index.md:35]: `openai-api-key` could be a setting or a config?

It's not safe to pass the key as a function argument.

**Comment by Xuanwo** [comment]
CI failed becuase `async-openai` doesn't has `rustls` support, please use the fork I created: https://github.com/64bit/async-openai/pull/60

File: docs/doc/15-sql-functions/61-ai-functions/index.md (hunk 2 above)
Comment (BohuTANG) [reply docs/doc/15-sql-functions/61-ai-functions/index.md:35]: Yes, we can do it later, for example: if the ai_to_sql argument only one, we can use the key from our config.  
The key can be revoked and re-create on openai platform.

**Comment by l1t1** [comment]
I tried the function in v1.0.23
```
Welcome to the MariaDB monitor.  Commands end with ; or \g.
Your MySQL connection id is 9
Server version: 8.0.26-v1.0.23-nightly-0e06b47048d90e53126f785e6db12c67cb592d46(rust-1.70.0-nightly-2023-03-18T16:36:53.222545484Z) 0

Copyright (c) 2000, 2018, Oracle, MariaDB Corporation Ab and others.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

MySQL [(none)]> CREATE DATABASE openai;
Query OK, 0 rows affected (0.079 sec)

MySQL [(none)]> USE openai;
Database changed
MySQL [openai]>
MySQL [openai]> CREATE TABLE users(
    ->     id INT,
    ->     name VARCHAR,
    ->     age INT
    -> );
Query OK, 0 rows affected (0.224 sec)

MySQL [openai]>
MySQL [openai]> SELECT * FROM ai_to_sql('List all users older than 30 years', '<openai-api-key>');
ERROR 1105 (HY000): Code: 1001, displayText = openai http error: reqwest::Error { kind: Builder, source: Custom { kind: NotFound, error: "error reading DNS system conf: No such file or directory (os error 2)" } }.
MySQL [openai]>
```

**Comment by BohuTANG** [comment]
@l1t1 Make your network can access openai

**Comment by l1t1** [comment]
database usually behind the firewall

**Pull request merged**